### PR TITLE
docs(api/filter): before_post_render & after_post_render process all files

### LINKED ISSUE #581: Mention hexo will transform every file in source/ unless excluded with skip_render.
See https://github.com/hexojs/hexo/issues/1591. We need to highlight the `skip_render` usage on these case. The `skip_render` has 110+ related issues on hexo, it is important to state that considering on our user profile.

### PR DESCRIPTION
**Pull request by curbengh**
Closes #581

## Check List

**Please read and check followings before submitting a PR.**

- [ ] I want to publish my theme on Hexo official website.
    - [ ] I have read the [theme publishing doc](https://hexo.io/docs/themes#Publishing).
    - [ ] `name` is unique.
    - [ ] `link` URL is correct.
    - [ ] `preview` URL is correct.
    - [ ] `preview` URL web site is rendered correctly.
    - [ ] Add a screenshot to `source/themes/screenshots`.
    - [ ] Screenshot filename is same as value of `name`.
    - [ ] Screenshot size is `800 * 500`.
    - [ ] Screenshot file format is `png`.
- [ ] I want to publish my plugin on Hexo official website.
    - [ ] I have read the [plugin publishing doc](https://hexo.io/docs/plugins#Publishing).
    - [ ] `name` is unique.
    - [ ] `link` URL is correct.
- [x] Others (Update, fix, translation, etc...)
    - Languages:
    - [x] `en` English
    - [x] `ko` Korean
    - [x] `pt-br` Brazilian Portuguese
    - [x] `ru` Russian
    - [x] `th` Thai
    - [x] `zh-cn` simplified Chinese
    - [x] `zh-tw` traditional Chinese

### CODE REVIEW

File: source/api/filter.md [hunk 1]
```diff
@@ -97,7 +97,18 @@ Executed before a post is rendered. Refer to [post rendering](posts.html#Render)
...
 
 ``` js
-hexo.extend.filter.register('before_post_render', function(data){
+hexo.extend.filter.register('before_post_render', function(data) {
+  data.title = data.title.toLowerCase();
+  return data;
+});
+```
+
+This filter matches every files in the `source/` folder, it may be necessary to skip non-markdown files, unless those files are listed in [`config.skip_render`](/docs/configuration#Directory); `themes/<your-theme>/source/` folder is not processed by this filter.
```
Comment (SukkaW) [inline source/api/filter.md:106]: Last time I checked, `before_post_render` only processes files under `source/_posts` directories.

Comment (curbengh) [reply source/api/filter.md:106]: I did a check before this PR, both `before_post_render` and `after_post_render` render everything in `source/`.

Comment (curbengh) [reply source/api/filter.md:106]: can confirm issue https://repl.it/repls/DefenselessHightechFormats#index.js

**Comment by stevenjoezhang** [comment]
Will be fixed in https://github.com/hexojs/hexo/pull/5118

**Pull request closed without merging**